src/Models/HydrostaticFreeSurfaceModels/vertical_vorticity_field.jl

```diff
@@ -0,0 +1,28 @@
+using Oceananigans.Grids: Face, Face, Center
+using Oceananigans.Operators: ζ₃ᶠᶠᵃ
+using Oceananigans.Fields: KernelComputedField, ComputedFieldBoundaryConditions
+
+using KernelAbstractions: @kernel
+
+@kernel function compute_vertical_vorticity!(ζ, grid, u, v)
+    i, j, k, = @index(Global, NTuple)
+    @inbounds ζ[i, j, k] = ζ₃ᶠᶠᵃ(i, j, k, grid, u, v)
+end
+
+"""
+    VerticalVorticityField(model)
+
+Returns a KernelComputedField that `compute!`s vertical vorticity in a
+manner consistent with the `VectorInvariant` momentum advection scheme
+for curvilinear grids.
+
+In particular, `VerticalVorticityField` uses `ζ₃ᶠᶠᵃ`, which in turn computes
+the vertical vorticity by first integrating the velocity field around the borders
+of the vorticity cell to find the vertical circulation, and then dividing by the area of
+the vorticity cell to compute vertical vorticity.
+"""
+function VerticalVorticityField(model)
+    return KernelComputedField(Face, Face, Center, compute_vertical_vorticity!, model, 
+                        boundary_conditions = ComputedFieldBoundaryConditions(model.grid, (Face, Face, Center)),
+                        computed_dependencies = (model.velocities.u, model.velocities.v))
+end
```

Comment:
Is this only consistent with `VectorInvariant` momentum advection? If so, maybe this constructor can throw an error or a warning if the model is not using `VectorInvariant`?

Reply:
Hmm, I'm not sure about that. I don't think it's "inconsistent" with the flux form equations, since vorticity does not appear in those. It could be consistent with a hypothetical advection scheme on a curvilinear grid that somehow uses vorticity, but a different vorticity than `VectorInvariant` uses.

On `RegularRectilinearGrid` there's no difference between this method of calculating vorticity and `∂x(v) - ∂y(u)`.

Perhaps we can use a more specific name, something like `CirculationVerticalVorticityField` to indicate that it's computed using circulation?

Reply:
I was just worried that the general name `VerticalVorticityField` might lead users to conclude that like e.g. `BuoyancyField` it works for all models. But sounds like this is fine for now since it actually does work for all models and advection schemes.

Reply:
@francispoulin do you want to visit that when `ShallowWaterModel` works with curvilinear grids? For rectilinear grids, you can write

```julia
∂x(vh / h) - ∂y(uh / h)
```

Reply:
That's a great idea.  I'm basically doing what you have above in the example, but I wonder if it might be more convenient to have something there already.